src/lib/patch.js

```diff
@@ -23,33 +23,36 @@ import cockpit from 'cockpit';
 
 export class Patch {
     static async patches() {
-        const xml = await cockpit.spawn(["zypper", "--xmlout", "list-patches"], { superuser : "require" });
-
-        const parser = new DOMParser();
-        const xmlDoc = parser.parseFromString(xml, "text/xml");
-
-        const updates = xmlDoc.getElementsByTagName("update");
-
-        return Array.from(updates).map((update) => {
-            return new Patch(
-                update.getAttribute("name"),
-                update.getAttribute("edition"),
-                update.getAttribute("category"),
-                update.getAttribute("severity"),
-                update.getElementsByTagName("summary")[0].textContent,
-                update.getElementsByTagName("description")[0].textContent
-            );
+        // variable for collecting the found patches
+        const patches = [];
+
+        const dbus_client = cockpit.dbus("org.opensuse.transactional_update");
+        const dbus_proxy = dbus_client.proxy("org.opensuse.transactional_update",
+                                             "/org/opensuse/transactional_update");
+
+        const transaction = (await dbus_proxy.call("CreateTransaction"))[0];
+        console.log("Created transaction: ", transaction);
+        const transaction_proxy = dbus_client.proxy("org.opensuse.transactional_update", transaction);
+
+        // register callbacks for the patch update details DBus signals,
+        // must be done before (!) calling the GetUpdates() DBus method
+        dbus_client.subscribe({ interface: "org.opensuse.transactional_update", path: transaction, member: "Update" },
+                              (path, iface, signal, args) => {
+                                  console.debug("Received patch: ", args[0]);
+                                  // collect the received patches
+                                  patches.push(Object.assign(new Patch(), args[0]));
```

Comment:
Why not simply `patches.push(new Patch(args[0]))`?

Reply:
Oh, I see, the Patch does not receive an object. But you could adapt it:

```js
constructor({name, version, category, severity, summary, description})
```

Reply:
I you have not done yet (perhaps you already did), I would suggest having a look at [Destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) chapter in MDN web docs.

Reply:
I know the `[foo, bar] = baz()` trick, I didn't know that it works for objects too...

Still a lot to learn. :smiley:

Reply:
Destructuring is a pretty powerful technique in JavaScript: you can set default values, rename variables, work with nested values, and so on. But the syntax can be puzzling sometimes if you are not used to it (at least for me). :smiley: